doc/callbacks.md

```diff
@@ -8,53 +8,74 @@ as:
 
 Callbacks can be passed as an argument to the `Optimize()` function:
 
+<details>
+<summary>Click to check out the example</summary>
```

Comment:
The `kramdown` Markdown processor, which is what's used for ensmallen.org, unfortunately doesn't work with this... for some reason the closing tag has to be on a separate line:

```
<summary>Click to check out the example
</summary>
```

That's kind of irritating, but, well, it's what we have to do I guess. :)

Reply:
:smiley: